src/backend/distributed/utils/colocation_utils.c

```diff
@@ -131,6 +187,31 @@ get_colocated_shard_array(PG_FUNCTION_ARGS)
 }
 
 
+static uint32
+CreateColocationGroupForRelation(Oid sourceRelationId)
+{
+	uint32 shardCount = ShardIntervalCount(sourceRelationId);
+	uint32 shardReplicationFactor = TableShardReplicationFactor(sourceRelationId);
+
+	Var *sourceDistributionColumn = DistPartitionKey(sourceRelationId);
+	Oid sourceDistributionColumnType = InvalidOid;
+	Oid sourceDistributionColumnCollation = InvalidOid;
+
+	/* reference tables has NULL distribution column */
+	if (sourceDistributionColumn != NULL)
+	{
+		sourceDistributionColumnType = sourceDistributionColumn->vartype;
+		sourceDistributionColumnCollation = sourceDistributionColumn->varcollid;
+	}
+
+	uint32 sourceColocationId = CreateColocationGroup(shardCount, shardReplicationFactor,
```

Comment:
We should try to move to a world where each colocation group has its own entry in pg_dist_colocation, but this is unfortunately not the case right now. Instead, pg_dist_colocation is a mapping which means as much as "if your settings match these values, use this as your default co-location group". 

When we're breaking co-location, all we really need to do is  call GetNextColocationId(), which will give us a unique co-location ID that we can set in pg_dist_partition. Other co-located tables will have the same co-location ID, without there being an entry in pg_dist_colocation.

Reply:
@marcocitus updated the PR.